### PR TITLE
Remove unused variable, fix typos

### PR DESCRIPTION
**Pull request by yowl**
This PR just cleans up an unused variable and some typos, found when adding the IL scanner to the LLVM backend.

### CODE REVIEW

File: src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/Target_X64/X64ReadyToRunHelperNode.cs [hunk 1]
```diff
@@ -150,7 +150,6 @@ protected override void EmitCode(NodeFactory factory, ref X64Emitter encoder, bo
                         }
                         else
                         {
-                            ISymbolNode targetMethodNode = target.GetTargetNode(factory);
```
Comment (MichalStrehovsky) [inline src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/Target_X64/X64ReadyToRunHelperNode.cs:153]: Could you do the same fix for Target_ARM/Target_ARM64? They all have this copy-pasted.

**Review by MichalStrehovsky — COMMENTED** [review]
Thank you!

File: src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/Target_X64/X64ReadyToRunHelperNode.cs (hunk 1 above)
Comment (yowl) [reply src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/Target_X64/X64ReadyToRunHelperNode.cs:153]: Righto, done.

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!

**Pull request merged**